lnpbp-0025.md

```diff
@@ -0,0 +1,117 @@
+```
+LNPBP: 0025
+Aliases: RGB25
+Vertical: Smart contracts
+Title: RGB collectible assets interface (RGB-25)
+Authors: Dr Maxim Orlovsky <orlovsky@lnp-bp.org>,
+         Zoe Faltibà,
+         Nicola Busanello,
+         Federico Tenga
+Comments-URI: <https://github.com/LNP-BP/LNPBPs/discussions>
+Status: Proposal
+Type: Standards Track
+Created: 2023-06-28
+Updated: 2023-06-28
+Finalized: ~
+Copyright: (0) public domain
+License: CC0-1.0
+```
+
+- [Abstract](#abstract)
+- [Background](#background)
+- [Motivation](#motivation)
+- [Design](#design)
+- [Specification](#specification)
+- [Compatibility](#compatibility)
+- [Rationale](#rationale)
+- [Reference implementation](#reference-implementation)
+- [Acknowledgements](#acknowledgements)
+- [References](#references)
+- [Copyright](#copyright)
+
+
+## Abstract
+
+
+## Background
+
+
+## Motivation
+
+
+## Design
+
+
+## Specification
+
+Interface specification is the following Contractum code:
+
+```haskell
+-- Defined by LNPBP-31 standard in `RGBContract.sty` file
+import juliet_resume_gilbert_6Hj3qNYF1PrUqoskKvvhGAyawPXXjYoAexEmMRBjQDV2 as RGBContract
+
+interface RGB25
+    global name :: RGBContract.Name
+    global details :: RGBContract.Details
+    global precision :: RGBContract.Precision
+
+    global data :: RGBContract.ContractData
+    global created :: RGBContract.Timestamp
+
+    -- State which contains amounts issued
+    global issuedSupply :: RGBContract.Amount
+
+    -- Ownership right over assets
+    private assetOwner* :: Zk64
+
+    genesis       :: name
+                   , details
+                   , precision
+                   , data
+                   , created
+                   , issuedSupply
+                   , reserves {RgbTypes.ProofOfReserves ^ 0..0xFFFF}
+                  -> assetOwner*
+                  -- errors which may be returned:
+                  !! supplyMismatch
+                   | invalidProof
+                   | insufficientReserves
+
+    op Transfer    :: previous assetOwner+
```

Comment:
Do we want to allow burning assets? Right now this is not possible in a way which can be detected at the contract level...

Reply:
Yes probably it's better to have burn feature, I'll add it

Reply:
Pls keep in mind that this will require counting the burned assets such that a total supply can be calculatable (the same way we do it in RGB20)